moshi/src/test/java/com/squareup/moshi/JsonValueSourceTest.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+import java.io.EOFException;
+import java.io.IOException;
+import okio.Buffer;
+import org.junit.Test;
+
+public final class JsonValueSourceTest {
+  @Test
+  public void simpleValues() throws IOException {
+    assertThat(jsonPrefix("{\"hello\": \"world\"}, 1, 2, 3")).isEqualTo("{\"hello\": \"world\"}");
```

Comment:
What do you think about making this function `splitJson` which returns `List<String>` so that we can assert the code does not consume too many bytes. This specific example would change to
```java
assertThat(splitJson("..")).containsExactly("json", ", 1, 2, 3");
```